nomenclature/codelist.py

```diff
@@ -388,6 +389,65 @@ def codelist_repr(self, json_serialized=False) -> Dict:
 
         return nice_dict
 
+    def filter(self, **kwargs) -> "CodeList":
+        """A filter that sorts a CodeList by any attribute
+
+        Parameters
+        ----------
+        self :
+        **kwargs :
+
+        Raises
+        ------
+        AttributeError
+            Provided attribute is not compatible with the given model.
+
+        Returns
+        -------
+        :class:'CodeList'
+
+        """
+
+        keep: List(Code) = []
+        for attribute in kwargs:
+            # Set the syntax as: code.<attribute>
+            for code in self.mapping.values():
+                if attribute in code.extra_attributes:
```

Comment:
I think you should be able to leave you this and the following three lines:
```suggestion
```
which means you should be good to leave out the entire `for code in self.mapping.values()`` loop.